src/components/Calculator.js

```diff
@@ -1,31 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Digits from './digits';
+import calculate from '../logic/calculate';
 
-const Calculator = () => (
-  <div className="main-calculator">
-    <div className="calculator-wrap">
-      <div className="calculator-result">
-        0
-      </div>
-      <div className="operation-container">
-        <div className="calculator-upper">
-          <div className="calculator-number">
-            <button type="button" alt="">AC</button>
-            <button type="button" alt="">+/-</button>
-            <button type="button" alt="">%</button>
-          </div>
-          <Digits />
+const Calculator = () => {
+  const [data, setData] = useState({
+    calc: {
+      total: null,
+      next: null,
+      operation: null,
+    },
+  });
+  const handleEvent = (e) => {
+    setData(({ calc }) => ({
+      calc: calculate(calc, e.target.value),
+    }));
+  };
+  return (
+    <div className="main-calculator">
+      <div className="calculator-wrap">
+        <div className="calculator-result">
+          {data.calc.total || ''}
+          {data.calc.operation || ''}
+          {data.calc.next || ''}
         </div>
-        <div className="calculator-operation">
-          <button type="button" alt="">÷</button>
-          <button type="button" alt="">x</button>
-          <button type="button" alt="">-</button>
-          <button type="button" alt="">+</button>
-          <button type="button" alt="">=</button>
+        <div className="operation-container">
+          <div className="calculator-upper">
+            <div className="calculator-number">
+              <button type="button" alt="" onClick={handleEvent} value="AC">AC</button>
+              <button type="button" alt="" onClick={handleEvent} value="+/-">+/-</button>
+              <button type="button" alt="" onClick={handleEvent} value="%">%</button>
```

Comment:
- [ ] The`%` button should work as modulus. In simpler terms, this is the remainder when dividing and should always be an integer. For instance `5 % 3` = 2 which means 2 is the remainder when you divide 5 by 3. Please implement this ♻️ 
You can see the result of the same operation in your app below 👇 
![image](https://github.com/freddykaberuka/mathematics/assets/46839548/67a4f58e-83b5-442f-86e9-2874c89ee177)